src/vs/base/browser/markdownRenderer.ts

```diff
@@ -150,8 +151,16 @@ export function renderMarkdown(markdown: IMarkdownString, options: MarkdownRende
 
 	// Will collect [id, renderedElement] tuples
 	const codeBlocks: Promise<[string, HTMLElement]>[] = [];
+	const syncCodeBlocks: [string, HTMLElement][] = [];
 
-	if (options.codeBlockRenderer) {
+	if (options.codeBlockRendererSync) {
+		renderer.code = (code, lang) => {
+			const id = defaultGenerator.nextId();
+			const value = options.codeBlockRendererSync!(postProcessCodeBlockLanguageId(lang), code);
+			syncCodeBlocks.push([id, value]);
```

Comment:
Can you write out the HTML directly here instead of adding it to `syncCodeBlocks`?

Reply:
And this API only lets me return html strings